.github/workflows/r.yml

```diff
@@ -52,6 +52,71 @@ env:
   DOCKER_VOLUME_PREFIX: ".docker/"
 
 jobs:
+  ubuntu-cpp-versions:
+    name: Arrow C++ Versions (${{ matrix.config.cpp_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        config:
+        - { cpp_version: "15.0.0" }
+        - { cpp_version: "14.0.2" }
+        - { cpp_version: "13.0.0" }
+
+    env:
+      ARROW_HOME: "${{ github.workspace }}/arrow"
+
+    steps:
+    - name: Checkout Arrow
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+      with:
+        path: src
+
+    - name: Cache Arrow C++ Build
+      id: cache-arrow-build
+      uses: actions/cache@v4
+      with:
+        path: arrow
+        key: arrow-r-cpp-release-${{ matrix.config.cpp_version }}-0
+
+    - name: Build Arrow C++ (${{ matrix.config.cpp_version }})
+      if: steps.cache-arrow-build.outputs.cache-hit != 'true'
+      run: |
+        curl -L "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${{ matrix.config.cpp_version }}/apache-arrow-${{ matrix.config.cpp_version }}.tar.gz" | \
+          tar -zxf -
+        mkdir arrow-build
+        export SOURCE_DIR="$(pwd)/arrow-apache-arrow-${{ matrix.config.cpp_version }}/cpp"
+        export DEST_DIR="$(pwd)/arrow"
+
+        src/r/inst/build_arrow_static.sh
```

Comment:
The caching works well in this PR but we will basically always have cache misses on main/other prs because we have such a large number of caches in the repo (most c++ related). I'd prefer precompiled binaries for that reason.

If we want to use the R specific one we can just use the ones in the artifactory for that: https://apache.jfrog.io/ui/repos/tree/General/arrow/r

Using the offical ubuntu ones like kou recommends might also be a good idea (as that is likely what cran will use for linux, no?